src/D2L.CodeStyle.Analyzers/Language/RequireNamedArgumentsAnalyzer.cs

```diff
@@ -63,29 +64,52 @@ SyntaxNodeAnalysisContext ctx
 			).ToImmutableArray();
 
 			if ( unnamedArgs.Length >= TOO_MANY_UNNAMED_ARGS ) {
-				// Pass the names and positions for each unnamed arg to the
-				// codefix.
-				var fixerContext = unnamedArgs.ToImmutableDictionary(
-					keySelector: binding => binding.Position.ToString(),
-					elementSelector: binding => binding.ParamName
-				);
-
 				ctx.ReportDiagnostic(
 					Diagnostic.Create(
 						descriptor: Diagnostics.TooManyUnnamedArgs,
 						location: expr.GetLocation(),
-						properties: fixerContext
+						properties: GetDiagnosticContext( unnamedArgs )
 					)
 				);
+
+				// Since this applies to all arguments and the suggested fix is
+				// to add names to all args we'll stop emitting diagnostics at
+				// this point.
+				return;
 			}
 
-			// TODO: literal args should always be named
+			foreach( var binding in unnamedArgs ) {
+				if ( binding.Syntax.Expression is LiteralExpressionSyntax ) {
+					ctx.ReportDiagnostic(
+						Diagnostic.Create(
+							descriptor: Diagnostics.LiteralArgShouldBeNamed,
+							location: expr.GetLocation(),
+							properties: GetDiagnosticContext( unnamedArgs ),
+							messageArgs: binding.ParamName
+						)
+					);
+
+					// Don't need any more diagnostics for this argument
+					continue;

```

Comment:
Similar to the previous comment, since the diagnostic and codefix is for the entirety of the call, this should be a return?

Reply:
I guess that's true ya